x-pack/plugins/observability_solution/apm/public/components/app/service_groups/service_group_save/group_details.tsx

```diff
@@ -31,22 +31,29 @@ interface Props {
   onClickNext: (serviceGroup: StagedServiceGroup) => void;
   onDeleteGroup: () => void;
   isLoading: boolean;
+  titleId?: string;
 }
 
+const DEFAULT_COLOR = '#5094C4';
+
 export function GroupDetails({
   isEdit,
   serviceGroup,
   onCloseModal,
   onClickNext,
   onDeleteGroup,
   isLoading,
+  titleId,
 }: Props) {
-  const [name, setName] = useState<string>(serviceGroup?.groupName || '');
+  const [name, setName] = useState(serviceGroup?.groupName);
   const [color, setColor, colorPickerErrors] = useColorPickerState(
-    serviceGroup?.color || '#5094C4'
+    serviceGroup?.color || DEFAULT_COLOR
   );
-
   const [description, setDescription] = useState<string | undefined>(serviceGroup?.description);
+
+  const isNamePristine = name === serviceGroup?.groupName;
```

Comment:
Sorry my ignorance, but what does `Pristine` mean? 😆

Reply:
@cauemarcondes] 
### Pristine Approach:
The pristine property of a form control indicates whether the control remains in its initial state. It is true when the control's value has not been changed, and false once the user modifies it. This property is useful for implementing conditional behavior—such as deciding when to show validation errors based on whether the user has interacted with the control or not.

### Why Was This Added?
We want to display validation errors only when the user interacts with specific form controls (e.g., clicking, typing) or presses the submit button. In this particular case, since the submit button is initially disabled, we rely on user interaction to trigger validation.

### Relation to Previous Comments
This is connected to your suggestion to use `useState(serviceGroup?.groupName ?? '')`. The problem with this approach is that it immediately triggers validation errors upon rendering the form, which is something we want to avoid. Cause `''===''` Our goal is to skip validation until the user interacts with the form.

As an alternative option, we could introduce pristine variables into the state to track whether each control has been touched, and update all callbacks accordingly. However, this approach seems a bit excessive and could add unnecessary complexity.

Reply:
Thanks for the thorough explanation.